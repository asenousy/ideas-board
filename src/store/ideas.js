import { createSlice } from "redux-starter-kit";
import { sanitize } from "dompurify";

const todoSlice = createSlice({
  initialState: [],
  reducers: {
    addIdea: (state, { payload }) => [...state, payload],
    updateIdea: (state, { payload }) =>
      state.map(idea =>
        idea.id === payload.id
          ? {
              id: idea.id,
              // sanitizing user input for xss,
              // even though react render does sanitize it,
              //but in case this data will be passed somewhere in future
              title: sanitize(payload.title),
              description: sanitize(payload.description),
              date: payload.date,
              new: false
            }
          : idea
      ),
    deleteIdea: (state, { payload }) =>
      state.filter(idea => idea.id !== payload.id)
  }
});

export default todoSlice;
