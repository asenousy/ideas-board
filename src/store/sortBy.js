import { createSlice } from "redux-starter-kit";

export const sortEnum = { date: "date", alphabetical: "alphabetical" };

const sortBy = createSlice({
  initialState: sortEnum.date,
  reducers: {
    setSortBy: (state, { payload }) => payload
  }
});

export default sortBy;
