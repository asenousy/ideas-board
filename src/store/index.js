import { combineReducers } from "redux";
import ideas from "./ideas";
import sortBy from "./sortBy";

export const reducer = combineReducers({
  ideas: ideas.reducer,
  sortBy: sortBy.reducer
});

export const actions = {
  ...ideas.actions,
  ...sortBy.actions
};

export { sortEnum } from "./sortBy";
