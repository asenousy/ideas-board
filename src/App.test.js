import React from "react";
import { Provider } from "react-redux";
import { configureStore } from "redux-starter-kit";
import { shallow, mount } from "enzyme";
import ConnectedApp, { App } from "./App";
import { reducer } from "./store";

const renderWithState = (Element, preloadedState) =>
  mount(
    <Provider store={configureStore({ reducer, preloadedState })}>
      <Element />
    </Provider>
  );

it("renders without crashing", () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});

it("creates Tiles in List", () => {
  const wrapper = renderWithState(ConnectedApp);
  const addButton = wrapper.find("Memo(AddTile)");
  expect(wrapper.find("Tile")).toHaveLength(0);
  addButton.simulate("click");
  expect(wrapper.find("Tile")).toHaveLength(1);
});

it("deletes Tiles from List", () => {
  const wrapper = renderWithState(ConnectedApp, {
    ideas: [
      {
        id: "1",
        title: "title",
        description: "description",
        date: 111,
        new: false
      }
    ]
  });
  expect(wrapper.find("Tile")).toHaveLength(1);
  const removeButton = wrapper
    .find("Tile")
    .find("FontAwesomeIcon")
    .at(1);
  removeButton.simulate("click");
  expect(wrapper.find("Tile")).toHaveLength(0);
});
