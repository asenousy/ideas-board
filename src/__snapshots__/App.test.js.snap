// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <UNDEFINED />,
        <div
          className="tiles"
        >
          <UNDEFINED
            onClick={[Function]}
          />
          <UNDEFINED />
        </div>,
      ],
      "className": "wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Header)",
          "type": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <UNDEFINED
              onClick={[Function]}
            />,
            <UNDEFINED />,
          ],
          "className": "tiles",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "type": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(TilesList)",
              "type": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <UNDEFINED />,
          <div
            className="tiles"
          >
            <UNDEFINED
              onClick={[Function]}
            />
            <UNDEFINED />
          </div>,
        ],
        "className": "wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(Header)",
            "type": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <UNDEFINED
                onClick={[Function]}
              />,
              <UNDEFINED />,
            ],
            "className": "tiles",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.memo),
                "compare": null,
                "type": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(TilesList)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
